### PR TITLE
WL-5290 Add copy link option to assignment list page.

### PR DESCRIPTION
**Pull request by RebeccaMiller-Which**

### CODE REVIEW

File: assignment/assignment-tool/tool/src/webapp/js/assignments.js [hunk 1]
```diff
@@ -963,3 +963,24 @@ ASN.handleReportsTriangleDisclosure = function (header, content)
...
+    $('#' + assignmentCount + '_copyLink').dialog("open");
+}
+
+//handles showing either the short url or the full url, depending on the state of the checkbox 
+// Copied from the portal module.
+ASN.toggleShortUrlOutput = function(defaultUrl, checkbox, textbox) {
+
+    if($PBJQ(checkbox).is(':checked')) {
+        $PBJQ.ajax({
+            url:'/direct/url/shorten?path='+encodeURI(defaultUrl),
```
Comment (buckett) [inline assignment/assignment-tool/tool/src/webapp/js/assignments.js]: I think this should be `encodeURIComponent` which will deal with the `&` issue

File: assignment/assignment-tool/tool/src/webapp/js/assignments.js [hunk 2]
```diff
@@ -726,10 +726,10 @@ ASN.invokeDownloadUrl = function(accessPointUrl, actionString, alertMessage, par
     if (document.getElementById('feedbackAttachments') && document.getElementById('feedbackAttachments').checked)
     {
         extraInfoArray[extraInfoArray.length]="feedbackAttachments=true";
-    }
```
Comment (buckett) [inline assignment/assignment-tool/tool/src/webapp/js/assignments.js]: It doesn't look like this block has changed.

File: assignment/assignment-tool/tool/src/webapp/js/assignments.js [hunk 3]
```diff
@@ -963,3 +963,24 @@ ASN.handleReportsTriangleDisclosure = function (header, content)
...
+{
+    $('#' + assignmentCount + '_copyLink').dialog("open");
+}
+
+//handles showing either the short url or the full url, depending on the state of the checkbox 
+// Copied from the portal module.
+ASN.toggleShortUrlOutput = function(defaultUrl, checkbox, textbox) {
+
+    if($PBJQ(checkbox).is(':checked')) {
+        $PBJQ.ajax({
```
Comment (buckett) [inline assignment/assignment-tool/tool/src/webapp/js/assignments.js]: $PBJQ should be $ as we're not in the portal.

**Review by buckett — CHANGES_REQUESTED** [review]

File: assignment/assignment-tool/tool/src/webapp/js/assignments.js (hunk 1 above)
Comment (RebeccaMiller-Which) [reply assignment/assignment-tool/tool/src/webapp/js/assignments.js]: Done.

File: assignment/assignment-tool/tool/src/webapp/js/assignments.js (hunk 2 above)
Comment (RebeccaMiller-Which) [reply assignment/assignment-tool/tool/src/webapp/js/assignments.js]: LF changed to CRLF, not sure why it is just for this block.  Can you fix pls?

File: assignment/assignment-tool/tool/src/webapp/js/assignments.js (hunk 3 above)
Comment (RebeccaMiller-Which) [reply assignment/assignment-tool/tool/src/webapp/js/assignments.js]: Changed.

**Review by RebeccaMiller-Which — COMMENTED** [review]
Also the short URL is automatically turned on now.

File: assignment/assignment-tool/tool/src/webapp/js/assignments.js [hunk 4]
```diff
@@ -964,23 +964,24 @@ ASN.handleReportsTriangleDisclosure = function (header, content)
     }
 }
 
-ASN.displayCopyLink = function(assignmentCount)
+ASN.displayCopyLink = function(assignmentCount, directUrl)
 {
+    $('.' + assignmentCount + '_urlholder').attr("value", ASN.toggleShortUrlOutput(directUrl, $('#directToolUrl'), assignmentCount + '_urlholder'));
```
Comment (buckett) [inline assignment/assignment-tool/tool/src/webapp/js/assignments.js]: You have multiple elements in the page that have `#directToolUrl` when you should only have one element on the page with that ID.
`ASN.toggleShortUrlOutput()` doesn't look to return anything so why are you setting the value of an element to it?

File: assignment/assignment-tool/tool/src/webapp/js/assignments.js (hunk 2 above)
Comment (buckett) [reply assignment/assignment-tool/tool/src/webapp/js/assignments.js]: Done.

File: assignment/assignment-tool/tool/src/webapp/js/assignments.js (hunk 3 above)
Comment (buckett) [reply assignment/assignment-tool/tool/src/webapp/js/assignments.js]: Magic, thanks.

File: assignment/assignment-tool/tool/src/webapp/js/assignments.js (hunk 4 above)
Comment (RebeccaMiller-Which) [reply assignment/assignment-tool/tool/src/webapp/js/assignments.js]: Changes made.

**Review by buckett — APPROVED** [review]

**Review by buckett — APPROVED** [review]

**Pull request merged**